qiskit/transpiler/passes/__init__.py

```diff
@@ -11,6 +11,7 @@
 from .cx_cancellation import CXCancellation
 from .fixed_point import FixedPoint
 from .dag_fixed_point import DAGFixedPoint
+from .adjacent_barriers import MergeAdjacentBarriers
```

Comment:
`MergeRedundantBarriers` might be clearer, since not all adjacent barriers will be merged.

Reply:
After some more thought, I think 'Adjacent' is better (since its possible there are redundant barriers in the DAG that won't be merged by this pass, at least not on a single run.) We should just be clear to document that only barriers which can be merged without affecting the barrier structure of the output DAG will be merged.

Reply:
I will add a comment to clarify this.

Reply:
I think something like this
```
qr = QuantumRegister(3, 'q')
        circuit = QuantumCircuit(qr)
        circuit.barrier(qr[0])
        circuit.barrier(qr[1])
        circuit.barrier(qr)
```

Reply:
can we call the file "merge_adjacent_barrier.py", to better match the pass name?